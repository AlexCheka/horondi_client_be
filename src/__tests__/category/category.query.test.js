/* eslint-disable no-undef */
const { gql } = require('apollo-boost');
const categoryService = require('../../modules/category/category.service')
const categoryResolver = require('../../modules/category/category.resolver')
const client = require('../../utils/apolloClient');

require('dotenv').config();


describe('querries', () => {
    test('should recive all categories like in service and resolver', async () => {
        const res = await client.query({
            query: gql`
                query {       
                    getAllCategories {
                        available
                        name {
                            value
                            lang
                        }
                        categoryCode
                        images {
                            large
                            medium
                            small
                            thumbnail
                        }
                    }
                }
            `
        })

        expect(categoryService.getAllCategories()).resolves.toBe(res);
    })

    test('should recive category by id like in service and resolver', async () => {
      const res = await client.query({
          query: gql`
              query {       
                getCategoryById(id: "5f0462f414438544ec76a778"){
                  _id
                  name {
                    value
                    lang
                  }
                  categoryCode
                  images {
                    large
                    medium
                    small
                    thumbnail
                  }
                }
              }
          `
      })

      expect(categoryService.getCategoryById("5f0462f414438544ec76a778")).resolves.toBe(res);
      expect(categoryResolver.categoryQuery.getCategoryById('',{id:"5f0462f414438544ec76a778"})).resolves.toBe(res);
      expect(res.data.getCategoryById).toMatchSnapshot()
  })
})