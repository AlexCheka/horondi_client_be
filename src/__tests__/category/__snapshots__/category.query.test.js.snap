// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`querries should recive category by id like in service and resolver 1`] = `null`;
